File: library/src/test/java/me/proxer/library/enums/SynonymTypeTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class SynonymTypeTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final SynonymType synonymType = api.moshi().adapter(SynonymType.class).fromJson("\"name\"");

        assertThat(synonymType).isSameAs(SynonymType.ORIGINAL);
    }

    @Test
    void testFallback() throws IOException {
        final SynonymType synonymType = api.moshi().adapter(SynonymType.class).fromJson("\"xyz\"");

        assertThat(synonymType).isSameAs(SynonymType.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/SynonymTypeTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class SynonymTypeTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(SynonymType::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"name\"") shouldBe SynonymType.ORIGINAL
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe SynonymType.OTHER
    }
}
